storage/storage.go

```diff
@@ -0,0 +1,78 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/stackpulse/steps-sdk-go/cloud/gcp"
+	"github.com/stackpulse/steps-sdk-go/log"
+	"gocloud.dev/blob"
+
+	_ "gocloud.dev/blob/gcsblob"
+	_ "gocloud.dev/blob/s3blob"
+)
+
+type Storage struct {
+	Bucket string `env:"BUCKET,required"`
+}
+
+func closeWithLog(closer io.Closer) {
+	if err := closer.Close(); err != nil {
+		log.Log("Failed to perform close: ", err)
+	}
+}
+
+// Init the environment and set up the relevant variables in order to run seamlessly.
+func (s *Storage) Init() error {
+	if isGCP, _ := gcp.IsGCPBucketURL(s.Bucket); isGCP {
+		// Note: our implementation injects the base64 representation of the credentials file from the integration.
+		// in order to seamlessly support it using go-cloud we will unpack the credentials to the expected format.
+		if _, err := gcp.SetAppCredentials(); err != nil {
+			return fmt.Errorf("set gcp credentials: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// UploadFile to the given bucket by where file is the local path and key is the desired remote path in the bucket.
+func (s *Storage) UploadFile(ctx context.Context, bucket *blob.Bucket, file, key string) error {
```

Comment:
```suggestion
func (s *Storage) Upload(ctx context.Context, file string, bucket *blob.Bucket, key string) error {
```
i think it makes more sense to pass file param before bucket as its upload from -> to so it should come first
also id call this Upload and the other function UploadFIles